shuup/admin/views/select.py

```diff
@@ -193,3 +200,68 @@ def _filter_query(self, request, cls, qs, shop, search_mode=None):
 
     def get(self, request, *args, **kwargs):
         return JsonResponse({"results": self.get_data(request, *args, **kwargs)})
+
+
+class ObjectSelectorView(TemplateView):
+    """
+    Base class for responding to searches from select2 components.
+    """
+
+    def get(self, request, *args, **kwargs):
+
+        selector = request.GET.get("selector")
+        search_term = request.GET.get("q").strip()
+        search_mode = request.GET.get("searchMode")
+        sales_units = request.GET.get("salesUnits")
+        user = request.user
+
+        shop = request.GET.get("shop")
+        if shop:
+            query_shop = Shop.objects.get_for_user(request.user).filter(pk=request.GET["shop"]).first()
+            if query_shop:
+                shop = query_shop
+        else:
+            shop = Shop.objects.get_for_user(request.user).first()
+
+        supplier = get_supplier(request)
+
+        if not (selector and search_term):
+            return JsonResponse({}, status=HTTPStatus.BAD_REQUEST)  # Error 400
+
+        for admin_object_selector_class in sorted(
+            get_provide_objects("admin_object_selector"), key=lambda provides: provides.ordering
+        ):
+            if not issubclass(admin_object_selector_class, BaseAdminObjectSelector):
+                continue
+
+            if not admin_object_selector_class.handles_selector(selector):
+                continue
+
+            admin_object_selector = admin_object_selector_class(selector, request=request)
```

Comment:
lets pass `shop`, `user` and `supplier` here. let's try to no depend on request now. check my comments below on the `BaseAdminObjectSelector` class.

Reply:
Good.